app/ota_client_stub.py

```diff
@@ -134,3 +206,246 @@ def _find_request(update_request, ecu_id):
             if request.ecu_id == ecu_id:
                 return request
         return None
+
+    def _update_executor(
+        self, entry, update_request: v2.UpdateRequest, *, pre_update_event: Event
+    ):
+        """
+        entry for local ota update
+
+        NOTE: no exceptions will be raised as there is no upper caller after the update is dispatched.
+        exceptions will only be recorded by logger. Use status API to query the status of the update.
+        """
+        post_update_event = Event()
+
+        # dispatch the local update to threadpool
+        _future = self._executor.submit(
+            self._ota_client.update,
+            entry.version,
+            entry.url,
+            entry.cookies,
+            pre_update_event=pre_update_event,
+            post_update_event=post_update_event,
+        )
+
+        # pulling subECU status
+        # NOTE: the method will block until all the subECUs' status are as expected
+        try:
+            asyncio.run(
+                self._ensure_subecu_status(
+                    timeout=server_cfg.WAITING_SUBECU_READY_TIMEOUT
+                )
+            )
+            # all subECUs are updated, now the ota_client can reboot
+            logger.debug("all subECUs are ready, set post_update_event")
+            post_update_event.set()
+
+            logger.debug("wait for local ota update to finish...")
+            exp = _future.exception(timeout=server_cfg.LOCAL_OTA_UPDATE_TIMEOUT)
```

Comment:
@Bodong-Yang I'll remove this timeout since the current implementation doesn't have a timeout.

Reply:
OK!